src/index.ts

```diff
@@ -40,9 +41,25 @@ async function asyncServerless() {
     });
 }
 
+async function checkPatientOrgsClaim(events: any) {
+    const decodedToken: any = jwt_decode(events.headers.Authorization.replace('Bearer ', ''));
+    if (
+        events.path === '/DetectedIssue' &&
+        events.httpMethod === 'GET' &&
+        decodedToken.patientOrgs &&
+        decodedToken.fhirUser &&
+        decodedToken.scp.some((scope: any) => scope.startsWith('user/'))
+    ) {
+        events.queryStringParameters = { 'patient:Patient.organization': decodedToken.patientOrgs };
+        events.multiValueQueryStringParameters = { 'patient:Patient.organization': [decodedToken.patientOrgs] };
+    }
+    return events;
+}
+
 const serverlessHandler: Promise<any> = asyncServerless();
 
 exports.handler = async (event: any = {}, context: any = {}): Promise<any> => {
+    event = await checkPatientOrgsClaim(event);
```

Comment:
Although what you have done is what is required, is there not a better place to add this, somewhere further up the stack, in serverlessHander?

Reply:
Unfortunately, this is the only place I could think of to intercept the event.  Yes, it will impact all the FHIR API requests to check the condition `checkPatientOrgsClaim`.

Reply:
Niroj, just curious if you considered implementing the change for this card in the fhir-works-on-aws-authz-smart module.  At first glance, that seems to be the module that would be most appropriate since it deals with request authorization.  Specifically, I noticed there was a function called [getSearchFilterBasedOnIdentity](https://github.com/lwarren01/fhir-works-on-aws-authz-smart/blob/mainline/src/smartHandler.ts#L186) there that looks like it inspects the claims in the access token and then return some type of "search filter" that is acted upon.  Again, _at first glance_, that seems like it might be a good spot for something like this.  

Also, a question.  If a Practitioner user makes a request to the the fhir API _without_ a patientOrgs claim, will they be returned an empty list, or will they be given admin access?  I can't tell from just looking at the code.

Reply:
yeah I did looked into fhir-works-on-aws-authz-smart repo, I was able to add patientOrgsClaim in userIdentity object. But But [authorizeAndFilterReadResponse](https://github.com/awslabs/fhir-works-on-aws-authz-smart/blob/mainline/src/smartHandler.ts#L325) hasAccessToResource method didn't help me filter the DetectedIssue resources I want. I intend to filter DetectedIssue resource using Organization ids in which the patient are referenced. For example here is the API call I want to filter when patientOrgsClaim is present:
{{API_URL}}/DetectedIssue?patient:Patient.organization=Organization/<org_id_1>,Organization/<org_id_2>

But the issue here is, DetectedIssue resource doesn't have reference to the organization directly and only has reference to the Patient resource. So I cannot use hasReferenceToResource function to filter the DetectedIssue resource with patientOrgsClaim.

I might have to look into making changes in [search component](https://github.com/awslabs/fhir-works-on-aws-search-es/tree/mainline/src) to get requried resoruces and than we can start filtering based on the patientOrgs claim. But I went with a hack method to penetrate the event in lambda instead. I will have a second look at the search component for sure.

And regarding you question about when patientOrgs claim is not present in the user, yeah they will return a empty list that means they are admin and will have access to all the resources mentioned in the scopes. For eg: the user will have all the access to all the detectedIssue resource if the user/DetectedIssue.read scope is present in the token and will not have access to any other resources other than DetectedIssue resource even if they are admin. Only system socpe have access to all the resources.

Reply:
The suggestion was to use the SearchFilter interface, which is returned by getSearchFilterBasedOnIdentity to return a filter representing the additional chained parameter: patient.organization. After looking a little deeper into how this interface is used in the search component (see [typeSearch](https://github.com/awslabs/fhir-works-on-aws-search-es/blob/mainline/src/elasticSearchService.ts#L140)), it doesn't look like we can use the existing interface for this additional query parameter. We will need to find an alternate method to feed in the additional query parameter.